scripts/vehicle_node.py

```diff
@@ -0,0 +1,380 @@
+        array_msg.layout.dim[0].size = 3
+        array_msg.layout.dim[0].stride = 3
+
+        return array_msg
+
+
+    def create_terrain_map(self):
+
+        #Get parameters
+        h = rospy.get_param('/gauss_h')
+        w = rospy.get_param('/gauss_w')
+        mu = rospy.get_param('/mu')
+        sigma = rospy.get_param('/sigma')
+        sigma_filt = rospy.get_param('/sigma_filt')
+        height = rospy.get_param('/height')
+        width = rospy.get_param('/width')
+        resolution =  rospy.get_param('/resolution')
+
+        #Fill array of size h x w with Gaussian Noise
+        grid_map = (h,w)
+        gaussian_array = np.random.normal(mu, sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, sigma_filt)
+
+        #Create an list and fill the list with x, y values and z the values will be the filtered gaussian noise
+        incr_x = 0
+        incr_y = 0
+        cloud = []
+        for x in np.arange(-height, (height+1), resolution):
+            for y in np.arange(-width, (width+1), resolution):
+                innerlist = []
+                innerlist.append(x)
+                innerlist.append(y)
+                innerlist.append(gaussian_array[incr_x][incr_y])
+                cloud.append(innerlist)
+                incr_y+=1
+            incr_x+= 1
+            incr_y = 0
+
+        return cloud
+
+
+    def velocity_cmd_callback(self, data):
+
+        #assign updated command velocities to twist variable
+        self.twist = data
+
+
+    def update_position(self, event):
+
+        resolution =  rospy.get_param('/resolution')
+        #time elapsed since last update position call (using event.current_real - event.last_real throws an error on first run as event.last_real does not exist in the first run)
+        time = 0.05
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = self.twist.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = self.twist.linear.x*time
+
+        #Calculate yaw, pitch and roll of the robot (pitch and roll currently not calculated)
+        self.pose.orientation.x = 0.0
+        self.pose.orientation.y = 0.0
+        self.pose.orientation.z += angle
+
+        #Calculate vehicle x, y, z position coordinates 
+        self.pose.position.x += (distance)*cos(self.pose.orientation.z)
+        self.pose.position.y += (distance)*sin(self.pose.orientation.z)
+
+        myarray = np.asarray(self.cloud)
+
+        points = []
+        values = []
+
+        for x in range(len(myarray)):
+            if (self.pose.position.x - 0.05 - resolution) <= myarray[x,0] <=  (self.pose.position.x + 0.05 + resolution):
+                if (self.pose.position.y - 0.05 - resolution) <= myarray[x,1] <= (self.pose.position.y + 0.05 + resolution):
```

Comment:
I'm not sure what the purpose of these loops is.  Add a comment.  @Kiran-H-github will likely have some feedback about how to do this more cleanly